x/ibc/07-tendermint/misbehaviour.go

```diff
@@ -59,8 +60,32 @@ func CheckMisbehaviourAndUpdateState(
 // checkMisbehaviour checks if the evidence provided is a valid light client misbehaviour
 func checkMisbehaviour(
 	clientState types.ClientState, consensusState types.ConsensusState, evidence types.Evidence,
-	height uint64, currentTimestamp time.Time,
+	height uint64, currentTimestamp time.Time, consensusParams *abci.ConsensusParams,
 ) error {
+	// calculate the age of the misbehaviour evidence
+	infractionHeight := evidence.GetHeight()
+	infractionTime := evidence.GetTime()
+	ageDuration := currentTimestamp.Sub(infractionTime)
+	ageBlocks := height - uint64(infractionHeight)
+
+	// Reject misbehaviour if the age is too old. Evidence is considered stale
+	// if the difference in time and number of blocks is greater than the allowed
+	// parameters defined.
+	//
+	// NOTE: The first condition is a safety check as the consensus params cannot
+	// be nil since the previous param values will be used in case they can't be
+	// retreived. If they are not set during initialization, Tendermint will always
+	// use the default values.
+	if consensusParams != nil &&
+		consensusParams.Evidence != nil &&
+		ageDuration > consensusParams.Evidence.MaxAgeDuration &&
+		ageBlocks > uint64(consensusParams.Evidence.MaxAgeNumBlocks) {
+		return sdkerrors.Wrapf(clienttypes.ErrInvalidEvidence,
+			"age duration (%s) and age blocks (%d) are greater than max consensus params for duration (%s) and block (%d)",
+			ageDuration, ageBlocks, consensusParams.Evidence.MaxAgeDuration, consensusParams.Evidence.MaxAgeNumBlocks,
+		)
+	}
+
 	// check if provided height matches the headers' height
 	if height > uint64(evidence.GetHeight()) {
```

Comment:
I don't quite understand what this is checking? If we want to make sure that the evidence isn't at a height in the future shouldn't we check the opposite i.e that if height < evidence.GetHeight()

Reply:
No, this `height` is the height from which we retrieve the old header to test from, so it must be before the two conflicting headers.